src/ucp/proto/proto.h

```diff
@@ -176,6 +176,7 @@ typedef struct {
                                                         protocol, for debugging */
 
     /* Output parameters */
+    const ucp_proto_t              *proto;      /* Pointer to protocol itself */
```

Comment:
can we avoid adding this?

Reply:
there are 2 options:
- change init() semantic adding this parameter
- calculate the result as a constant manually for every protocol, but I don't like this approach because of human factor

Reply:
2nd option means we may have to check errh requirement in every protocol, but IMO some protocols checks could be united (such as rma/amo/tag offload), and also some rndv protocols have to implement err handling anyway. So maybe it won't be in so many places in the code.

Reply:
to make it in runtime, we need to a have some parameter with access to `abort` function pointer, otherwise the initialization path must be checked for every protocol manually, IMO it's a bug-prone way

Reply:
not sure that presence of "abort" is right indication of errh support.
maybe the protocol is such that abort is not even needed in order to handle errors?

Reply:
then it would be `empty_func`, anyway we call `proto.abort` on error flow in many paces